### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by jaepark98**
This is a Doxyfile and README.md update PR. The functions have been changed to be able to provide documentation.

To create documentation, just run `doxygen` inside the docs directory. The html and latex folders, being generated as a result of this command, have not been added.

Additionally, to prepare for the last PR, the large dictionary that causes server crash has been deleted. dictionary0.txt remains.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,90 +4,112 @@
 Spellcheck is a tool which, given a text, will search for misspelled words and will suggest alternative spellings.
 
 ## 2. Compilation Information
-Right now we have a lot of Makefiles. We should finalize compilation by editing those - or at least indicate shared libraries here
+### Setup
+For the API setup, please refer to [Redis and Hiredis installation](https://github.com/cmsc22000-project-2018/api). 
 
-## 3. Interactive Mode Description
+### Compilation
+To use spellcheck, run `make` in the root directory.
```
Comment (dwahme) [inline README.md]: To "build" spellcheck

File: README.md [hunk 2]
```diff
@@ -4,90 +4,112 @@
...
 ```
-Line 32:
-and then the fox jomped over the fens
-                 ^^^^^^
-Edit the specified misspelling? [y/n] 
-(You can press [r] to return to the home interface)
-spellcheck >
+Line: 1:
+I have a splling checker
+            ^^^^^^^
```
Comment (dwahme) [inline README.md]: formatting- underline doesn't match

File: README.md (hunk 1 above)
Comment (jaepark98) [reply README.md]: Fixed!

File: README.md (hunk 2 above)
Comment (jaepark98) [reply README.md]: Fixed!

**Pull request merged**